src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList(int initCapacity) {
+        if (initCapacity < 0) {
+            throw new ArrayListIndexOutOfBoundsException("this index is invalid");
+        }
+        values = (T[]) new Object[MAX_ARRAY_LENGTH];
+    }
+
+    public ArrayList() {
+        this(MAX_ARRAY_LENGTH);
+    }
+
+    private void resizeArray() {
+        if (size == values.length) {
+            int sizeNewArray = (int) (values.length * 1.5);
+            T[] newArray = (T[]) new Object[(int) (sizeNewArray)];
+            System.arraycopy(values, 0, newArray, 0, size);
+            values = newArray;
+        }
+    }
+
+    private void checkIndex(int index, int size) throws ArrayListIndexOutOfBoundsException {
```

Comment:
you don't need to add parameter size in that method, we've already know size for our list